src/components/BookDetail.js

```diff
@@ -1,71 +1,31 @@
-import React, { useState } from "react";
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { remveBook } from "../redux/books/booksSlice";
 import CompletionCircle from "./CompletionCircle";
 
 const BookDetail = () => {
-  const booksDB = [
-    {
-      id: 1,
-      type: "Action",
-      bookName: "The Hunger Games",
-      author: "Suzanne Collins",
-      CommentsBtn: "Comments",
-      RemoveBtn: "Remove",
-      EditBtn: "Edit",
-      Percentage: "64",
-      Complete: "Complete",
-      CurrentChapter: "CURRENT CHAPTER",
-      Chapter: "Chapter 17",
-      UpdateProgress: "UPDATE PROGRESS",
-    },
-    {
-      id: 2,
-      type: "Science Fiction",
-      bookName: "Dune",
-      author: "Frank Herbert",
-      CommentsBtn: "Comments",
-      RemoveBtn: "Remove",
-      EditBtn: "Edit",
-      Percentage: "8",
-      Complete: "Complete",
-      CurrentChapter: "CURRENT CHAPTER",
-      Chapter: 'Chapter 3: "A Lesson Learned"',
-      UpdateProgress: "UPDATE PROGRESS",
-    },
-    {
-      id: 3,
-      type: "Economy",
-      bookName: "Capital in the Twenty-First Century",
-      author: "Suzanne Collins",
-      CommentsBtn: "Comments",
-      RemoveBtn: "Remove",
-      EditBtn: "Edit",
-      Percentage: "0",
-      Complete: "Complete",
-      CurrentChapter: "CURRENT CHAPTER",
-      Chapter: "Introduction",
-      UpdateProgress: "UPDATE PROGRESS",
-    },
-  ];
-  const [books, setBooks] = useState(booksDB);
-  const handleDeleteBook = (id) => {
-    setBooks((oldBooks) => {
-      return oldBooks.filter((theBook) => theBook.id !== id);
-    });
+  const books = useSelector((state) => state.books);
+  const dispatch = useDispatch();
+  // const [books, setBooks] = useState(booksDB);
```

Comment:
**Great job!**

- [ ] Books is loading as expected. However, your remove functionality of your app is not working. It's showing an error. Please fix the issue and make sure app is running as expected!
 <img width="836" alt="Screenshot 2023-06-13 at 18 49 19" src="https://github.com/DilsherB/bookstore/assets/69011963/c3395a86-4299-46cc-bc8a-2d9cfd960b75">

Reply:
@Rustamxon7
Thank you for your detailed review feed back. 
I tried a much to implement remove functionality but couldn't. Please help me if you can.